Journalator/Source/InvestONator.lua

```
@@ -0,0 +1,391 @@
+    return false
+  end
+  
+  local portfolio = JOURNALATOR_INVEST_O_NATOR_DATA.portfolios[portfolioId]
+  portfolio.items = portfolio.items or {}
+
+  -- Determine final item ID:
+  -- - If caller provides a non-positive/sentinel value, generate a unique synthetic ID.
+  -- - If caller provides an existing positive ID that already exists, do not overwrite.
+  local finalItemId = tonumber(itemId)
+  if not finalItemId or finalItemId <= 0 then
+    local candidate = portfolio.nextSyntheticItemId or -1
```

Luacheck (1.2.0)

[warning] line is too long (124 > 120)

(W631)